packages/framework/aqueduct/src/data-objects/dataObject.ts

```diff
@@ -62,8 +62,10 @@ export abstract class DataObject<O extends IFluidObject = object, S = undefined,
      * Initializes internal objects and calls initialization overrides.
      * Caller is responsible for ensuring this is only invoked once.
      */
-    public async initializeInternal(): Promise<void> {
-        if (!this.runtime.existing) {
+    public async initializeInternal(existing?: boolean): Promise<void> {
+        const backCompatExisting = instantiateExisting(this.context, existing);
```

Comment:
I'm not sure this is required. this method is called only by finishInitialization() that passes boolean (no undefined). Tracing code up, you already do instantiateExisting() dance in instantiateDataStore(), so all the code down operates with boolean

Reply:
This is the end of the (long) story of my back-compat testing. I'll check to see if I can simplify it. Thanks!